tools/lib/NQP/Config/Rakudo.pm

```diff
@@ -612,6 +611,7 @@ sub gen_nqp {
     for my $b ( $self->active_backends ) {
         my $bconfig = $self->backend_config($b);
         my $postfix = $self->backend_abbr($b);
+        my $exe     = $b eq 'moar' ? $config->{exe} : $config->{bat};
```

Comment:
I don't like us being too specific here. Consider rakudo getting more backends with different runner extensions (a voice inside me screams: "Damn you, Bill Gates!").

Perhaps, the most correct solution would be to set `nqp_runner` config variable in `configure_<backend>_backend` methods. To make life a little bit easier we might set it to some kind of common default – say, to `File::Spec->catfile( $prefix, 'bin', "nqp-$postfix$bat" );` so that backends with exe would just have to override it with `File::Spec->catfile( $prefix, 'bin', "nqp-$postfix$exe" );`. In other words, in the current situation it would mean that only `configure_moar_backend` would have to override it. Then at this location we would just use `$bin = $config->{nqp_runner}`.

Alternatively, we could iterate over exe and bat extensions and test for executable existence. But this method is slower and prone to errors if a runtime switches from script to binary format.

BTW, I just have noticed that `$tpath` from the next line is never used. Perhaps it's a leftover of some older or even debug code.